### PR TITLE
fix: identify traits type

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-js-common/src/types/traits.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  id?: string;
+  firstName?: string;
+  lastName?: string;
+  name?: string;
+  age?: number;
+  email?: string;
+  phone?: string;
+  address?: string;
+  birthday?: string;
+  company?: string;
```
Comment (bardisg) [inline packages/analytics-js-common/src/types/traits.ts:16]: this is a object in our docs

File: packages/analytics-js-common/src/utilities/eventMethodOverloads.ts [hunk 2]
```diff
@@ -23,7 +24,7 @@ export type TrackCallOptions = {
 
 export type IdentifyCallOptions = {
   userId?: string | null;
-  traits?: Nullable<ApiObject>;
+  traits?: Nullable<IdentifyTraits>;
```
Comment (bardisg) [inline packages/analytics-js-common/src/utilities/eventMethodOverloads.ts:27]: do group traits forllow the same object type?

**Review by bardisg — COMMENTED** [review]
Please also check for 'traits.' and 'traits?' in the integration package src folder.

Fill in the type with any properties that are used in our integrations but are not documnets in the docs.

After we finish we have to update the docs too

File: packages/analytics-js-common/src/utilities/eventMethodOverloads.ts (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-js-common/src/utilities/eventMethodOverloads.ts:27]: Not necessarily. It is about the group information where, firstName, dob doesn't make sense.

File: packages/analytics-js-common/src/types/traits.ts (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-common/src/types/traits.ts]: Adressed

File: packages/analytics-js-common/src/types/traits.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ */
+export type IdentifyTraits = {
+  id?: string;
+  firstName?: string;
+  lastName?: string;
+  name?: string;
+  age?: number;
+  email?: string;
+  phone?: string;
+  address?: string;
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply packages/analytics-js-common/src/types/traits.ts]: Adressed

**Review by bardisg — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Review by bardisg — APPROVED** [review]

**Pull request merged**